src/Raven.Server/Documents/Handlers/Debugging/ThreadsHandler.cs

```diff
@@ -24,11 +24,19 @@ public Task StackTrace()
             var threadIds = GetStringValuesQueryString("threadId", required: false);
             var includeStackObjects = GetBoolValueQueryString("includeStackObjects", required: false) ?? false;
 
-            using (var sw = new StreamWriter(ResponseBodyStream(), Encoding.UTF8,1024, true))
+            using (var stream = new MemoryStream())
             {
-                OutputResultToStream(sw, threadIds.ToHashSet(), includeStackObjects);
+                using (var sw = new StreamWriter(stream))
+                {
+                    OutputResultToStream(sw, threadIds.ToHashSet(), includeStackObjects);
+                 
+                    sw.Flush();
+                    
+                    stream.Position = 0;
```

Comment:
here I wanted to avoid case when we start writing to response - and HTTP 200 was sent, then we encounter an error. Right now we either send error (with 5xx status) which is valid JSON - or valid response.